src/models/CommonModel.ts

```diff
@@ -107,11 +123,12 @@ export class CommonModel extends CommonSchema<CommonModel> {
       }
     }
     if (this.properties !== undefined && Object.keys(this.properties).length) {
-      Object.entries(this.properties).forEach(([, propertyModel]) => {
-        if (propertyModel.$ref !== undefined) {
-          dependsOn.push(propertyModel.$ref);
-        }
-      });
+      const refProperties = (propertyModel: CommonModel) => propertyModel.$ref !== undefined;
+      const refProperty = (propertyModel: CommonModel) => `${propertyModel.$ref}`;
+      const referencedProperties = Object.values(this.properties)
+        .filter(refProperties)
+        .map(refProperty);
```

Comment:
Really, please add these function inside `fliter` and `map` 😄

Reply:
@jonaslagoni  @magicmatatjahu  It is not bad at all. It is actually good practice but you need to move them at the top of file and export them. In this way you could unit test those functions, it is also tiny step into functional programming.